spring-boot-actuator/src/main/java/org/springframework/boot/actuate/autoconfigure/HealthIndicatorAutoConfiguration.java

```diff
@@ -363,4 +392,51 @@ protected ElasticsearchHealthIndicator createHealthIndicator(Client client) {
 
 	}
 
+	/**
+	 * A {@link BeanPostProcessor} that registers {@link HealthIndicator} beans with a
+	 * {@link HealthIndicatorRegistry}.
+	 */
+	private static class HealthIndicatorRegistryPostProcessor implements BeanPostProcessor {
```

Comment:
It makes sense to have a single entry point for registering ```HealthIndicator``` instances, doesn't it? Also using this approach if I users provide their own implementation of ```HealthIndicatorRegistry``` they would automatically get ```HealthIndicator``` Spring beans registered. Finally, backwards compatibility with [current naming convention](http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#_writing_custom_healthindicators) should be preserved:
> The identifier for a given ```HealthIndicator``` is the name of the bean without the ```HealthIndicator``` suffix if it exists. In the example above, the health information will be available in an entry named ```my```.